src/scss/share.scss

```diff
@@ -98,22 +87,54 @@
 ///
 /// @param {String} $name - Name of the icon
 /// @param {String} $classname [o-share] - base class name
-@mixin oShareActionIcon($name, $classname: o-share) {
-	.#{$classname}__action--#{$name} i {
-		background-image: url(_oShareGetIconUrl($name, $format: png) + "&width=" + _oShareStripUnits($o-share-icon-size));
-		// The weird value for rgba is to prevent Sass converting rgba(0,0,0,0) to 'transparent',
-		// which would cause the following line of css to be understood by ie7/8 and the svg to be downloaded
-		background: rgba(0, 0, 0, 0.0001) url(_oShareGetIconUrl($name)) center/100% no-repeat;
+@mixin oShareActionIcon($icon-name, $classname: o-share) {
+	// sass-lint:disable-all
+	$scheme: "ftsocial-v#{$o-share-social-icons-version}";
+
+	@if ($icon-name == 'mail' or $icon-name == 'link') {
+		$scheme: "fticon-v#{$o-share-ft-icons-version}";
+	}
+
+	$service-url: "#{$o-share-image-service-base-url}/#{$o-share-image-service-version}/images/raw/#{$scheme}:#{$icon-name}";
+	$query: "?source=o-share";
+
+	$hover-color: oColorsGetColorFor('o-share-button-hover', background);
+
+	$hover-color: str-slice(ie-hex-str($hover-color), 4);
+	$tint: "&tint=%23#{$hover-color},%23#{$hover-color}";
+
+	// scss-lint:disable DuplicateProperty
+	.#{$classname}__action--#{$icon-name} i {
+		background-image: url($service-url + $query + "&format=svg");
+
+		// ie7/8 fallback. Uses the `\9` selector hack to target ie6-8.
+		// Hack is documented here: http://browserhacks.com/#hack-ab1bcc7b3a6544c99260f7608f8e845e
+		// It still needs to use the build service  <-- what does this comment mean?
```

Comment:
Erm yes I have and no I don't know what it means either. Shall I just remove that line?